ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/healthcheck/HealthCheckAgent.java

```diff
@@ -84,7 +87,15 @@ public HealthCheckResponse checkHealth() {
         ));
     final boolean allHealthy = results.values().stream()
         .allMatch(HealthCheckResponseDetail::getIsHealthy);
-    return new HealthCheckResponse(allHealthy, results);
+    final Map<String, Set<TaskMetadata>> metadata = ksqlClient
```

Comment:
Could you implement this check in another class, i.e. QueryUpstreamCheck? Like we do with the metastore, kafka, and commandrunner checks. It is more structured and won't require changes in HealthCheckResponse.

Reply:
I did do that at first but that would require a change to the HealthCheckResponseDetail which then would complicate all the other checks. If you think that is better we can do that instead though